sync3/handler/connstate.go

```diff
@@ -87,10 +95,13 @@ func NewConnState(
 //   - load() bases its current state based on the latest position, which includes processing of these N events.
 //   - post load() we read N events, processing them a 2nd time.
 func (s *ConnState) load(ctx context.Context, req *sync3.Request) error {
-	initialLoadPosition, joinedRooms, joinTimings, err := s.globalCache.LoadJoinedRooms(ctx, s.userID)
+	initialLoadPosition, joinedRooms, joinTimings, loadPositions, err := s.globalCache.LoadJoinedRooms(ctx, s.userID)
 	if err != nil {
 		return err
 	}
+	for roomID, nid := range loadPositions {
+		s.loadPositions[roomID] = nid
+	}
```

Comment:
Any reason for doing this over `s.loadPositions = loadPositions`?

Reply:
Nope, let's assign instead.